### PR TITLE
Add landscape time picker

### LINKED ISSUE #988: Missing landscape layout of time pickers
screenshot_20151217-153210

### PR DESCRIPTION
**Pull request by abarth**
Fixes #988

### CODE REVIEW

**Comment by abarth** [comment]
@HansMuller @Hixie @cbracken

File: packages/flutter/lib/src/material/time_picker.dart [hunk 1]
```diff
@@ -35,6 +40,18 @@ class TimeOfDay {
   /// argument must be between 0 and 59, inclusive.
   const TimeOfDay({ @required this.hour, @required this.minute });
 
+  /// Creates a time of day based on the given time.
+  ///
+  /// The [hour] is set to the time's hour and the [minute] is set to the time's
+  /// minute.
```
Comment (Hixie) [inline packages/flutter/lib/src/material/time_picker.dart]: in which time zone?

File: packages/flutter/lib/src/material/time_picker.dart [hunk 2]
```diff
@@ -35,6 +40,18 @@ class TimeOfDay {
...
+  TimeOfDay.fromDateTime(DateTime time) : hour = time.hour, minute = time.minute;
+
+  /// Creates a time of day based on the current time.
+  ///
+  /// The [hour] is set to the current hour and the [minute] is set to the
+  /// current minute.
```
Comment (Hixie) [inline packages/flutter/lib/src/material/time_picker.dart]: in which time zone?

File: packages/flutter/lib/src/material/time_picker.dart [hunk 3]
```diff
@@ -205,10 +156,22 @@ class _TimePickerHeader extends StatelessWidget {
     onChanged(selectedTime.replacing(hour: newHour));
   }
 
+  TextStyle _getBaseHeaderStyle(TextTheme headerTextTheme) {
+    assert(orientation != null);
+    switch (orientation) {
+      case Orientation.portrait:
+        return headerTextTheme.display3.copyWith(fontSize: 60.0);
```
Comment (Hixie) [inline packages/flutter/lib/src/material/time_picker.dart]: are these numbers from the spec? you should add a comment justifying these numbers.

Comment (abarth) [inline packages/flutter/lib/src/material/time_picker.dart]: The spec doesn't say what sizes to use.  If we use these sizes, we get pixel identical results to the screenshots in the spec.  /me will document.

**Comment by Hixie** [comment]
LGTM

**Pull request merged**